### DISCUSSION TITLE
Current state node not working after a device trigger?

### DISCUSSION THREAD

**Discussion by selmaohneh** [zachowj/node-red-contrib-home-assistant-websocket#510] [Q&A] [answered]
Here a screenshot of a flow of mine:

![image](https://user-images.githubusercontent.com/29881939/141102112-6eb2502b-9273-4290-83d9-dff04bbe213b.png)

As you can see, when I trigger the flow via the inject node, both current state nodes do work as expected, i.e. they check the luminosity of the motion sensors. The correct luminosity value gets shown below the node.

When I invoke the flow with the device trigger, i.e. when I walk on the upper hallway, the first current state node right after the trigger seems to not work. It shows NaN for some reason.

![image](https://user-images.githubusercontent.com/29881939/141102512-02a558f1-c006-4335-a9f0-921adaf361b3.png)

This behaviour is reproducable for me. I get the same results every time, event after NodeRed or Hassio restarts.

Here the content of the involved nodes:

![image](https://user-images.githubusercontent.com/29881939/141102761-3a0d3564-6601-46c9-89f1-cb6d76131ceb.png)
![image](https://user-images.githubusercontent.com/29881939/141102786-77d8a8bf-1216-47e3-838c-571fa34e05fb.png)
![image](https://user-images.githubusercontent.com/29881939/141102831-7dda1524-f3ce-45f9-9f64-9ae30f8ce95c.png)

Any ideas? Is this a bug or am I doing something wrong?

Thanks!

**Comment by zachowj** [comment, marked as answer]
The currentv state node is probably getting overwritten by the payload. On the current state of node check block input overrides

**Reply by selmaohneh** [reply]
That's it, thanks!
